aten/src/ATen/core/ivalue.cpp

```diff
@@ -634,21 +634,47 @@ CAFFE2_API intrusive_ptr<ivalue::Future> collectAll(
   return ctx->dstFuture;
 }
 
-CAFFE2_API intrusive_ptr<ivalue::Future> collectAll(
-    std::vector<intrusive_ptr<ivalue::Future>> srcs) {
-  auto typePtr = srcs.empty() ? AnyType::get() : srcs[0]->elementType();
-  // Lists are generally expected to have a consistent type, check
-  // for this unless we have a compelling target use case.
-  for (size_t i = 1, len = srcs.size(); i < len; ++i) {
```

Comment:
I think we should still those torch check logics in the `collectAll` c10::List overload, to ensure the single type list constraint.

Reply:
Sounds good,  adding a `TORCH_CHECK` to the `addCallback` loop.